File: Ruby/challenge21.rb
def sum_integer(n)
  # Split the number into digits and use inject with a `memo`
  return n.split(//).inject(0) { |sum, n| sum + n.to_i }
end

File.open(ARGV[0]).each_line do |line|
  puts sum_integer(line.strip())
end

File: Python/challenge21.py
import sys

def sum_integer(n):
    """Returns the sum of the digits in an integer."""
    return sum([int(elem) for elem in str(n)])

test_cases = open(sys.argv[1], 'r')

for test in test_cases:
    print sum_integer(test.strip())

test_cases.close()
